### ISSUE TITLE
Connection is never released after multiple messages from vertx-blocked-thread-catcher

### ISSUE THREAD

**Issue by dmytro-kostiushko** [hibernate/hibernate-reactive#1431]
Describe the bug
We have had a bug with our system that led to thousands of elements saved in the mongodb document, which in turn made any fetch request overflow the allowed time of execution that is expected by vertx-blocked-thread-catcher, but the issue is that after a hundred or so of messages like this our system shuts down entirely even the most basic functions in the app like find by id no longer respond and only produce a io.vertx.core.impl.NoStackTraceThrowable: Timeout exception. Nowhere in the application are we are using any blocking calls, we are utilizing hibernate reactive in combination wit spring webflux and spring data mongodb reactive

Expected behavior
The connection to the db is released after the thread that is holding it has finished its execution

Actual behavior
The connection is never released leading to entire application shutdown

How to Reproduce?
Create a placeholder application that takes more time to process a request then it is allowed by vertx-blocked-thread-checker spam the app with hundred or so requests, then after that is complete attempt sending a request should be processed in time that is less then "vertx-blocked-thread-checker", observe that the latter request is failing

Output of uname -a or ver
No response

Output of java -version
11.4

GraalVM version (if different from Java)
No response

Quarkus version or git rev
Spring Webflux is used.

Build tool (ie. output of mvnw --version or gradlew --version)
Gradle 7.5.1

Additional information
No response

**Comment by DavideD** [comment]
Is this the right issue tracker? MongoDB is not among the supported databases

**Comment by dmytro-kostiushko** [comment]
The issue is not with the mongodb, it is just my understanding that because server is so hevly loaded by fetching a big  record from the mongodb,  vertx-blocked-thread-catcher is precieving this as a blocked thread, but the issue is not with this, but rather with the fact that after the query fails because of the timeout, it looks like hibernate does not release the connection back into the pool. After which even when the load on the server subsides all queries continue to fail regardles

**Comment by DavideD** [comment]
OK, I think I get it now.
Is there anything in the log by any chance?

**Comment by dmytro-kostiushko** [comment]
Sorry for the late response, not really, I only get the only stack trace I am getting is 

```
io.vertx.core.impl.NoStackTraceThrowable: Timeout | io.vertx.core.impl.NoStackTraceThrowable: Timeout
-- | --
```

Maybe I need to change the logging level to see more, but not sure what package I should target

**Comment by agreedSkiing** [comment]
Hi,

I’m also starting to get this error now that I’m upgrading our application to quarkus 3.3.1 from 2.16.8.Final and this error is an intermittent one since it sometimes happens when our cronjobs starts multiple database requests in our tests suit. Or well… to be more specific again, the tests that are reliant on the cronjobs run in a profile that triggers the two cronjobs every second and some of these tests manually start cronjob method to ensure a more reliable test.

I have also tried to upgrade to quarkus 3.1.1.Final since that contains the 2.0.0.Final release of hibernate reactive but to no avail.

Below are some logs from the failures with the quarkus application log level set to trace and the application properties used for the database connection and also the java code that currently throws the stacktrace with its calling method.

## Properties

```properties
quarkus.hibernate-orm.jdbc.timezone = UTC
quarkus.datasource.db-kind = postgresql
quarkus.datasource.reactive.url = vertx-reactive:postgresql://localhost:5432/quarkus
quarkus.datasource.username = user
quarkus.datasource.password = pass
quarkus.datasource.jdbc.url = jdbc:postgresql://localhost:5432/quarkus
quarkus.datasource.jdbc = false #Using flyway implementation https://github.\
    com/quarkusio/quarkus/issues/10716#issuecomment-1069943206
```

## Java code

### Cronjob method (cronjob class)

```java
    @Scheduled(delayed = "10s", every = "{reindex.addresses.schedule.every}")
    public Uni<Void> runTask() {
        log.debugv("AddressReindexScheduler start");
        reindexService
            .processBatch()
            .subscribe()
            .with(
                item -> log.debugv("AddressReindexScheduler subscribe: {0}", item.size()),
                failure -> {
                    log.error("AddressReindexScheduler failure", failure);
                    throw new InternalServerErrorException(failure);
                }
            );
        return Uni.createFrom().voidItem();
    }
```

### Calling method (service class)

```java
    public Uni<List<AddressReindex>> processBatch() {
        log.fatal("##### processBatch");
        return reindexRepository
            .getBatch(100)
            .invoke(() -> log.fatal("########## GOT BATCH"))
            .onItem()
            .transformToMulti(list -> Multi.createFrom().iterable(list))
            .onItem()
            .call(this::processItem)
            .collect()
            .asList();
    }
```

### DB method (repository class)

```java
    public Uni<List<AddressReindex>> getBatch(int max) {
        String query = "SELECT a FROM AddressReindex a";
        return sf
            .withSession(session ->
                session.createQuery(query, AddressReindex.class).setFirstResult(0).setMaxResults(max).getResultList()
            )
            .onFailure()
            .transform(IllegalStateException::new);
    }
```

## Quarkus 3.3.1.

```log
2023-09-08 08:01:48,789 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (main) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:49,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:49,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:49,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:50,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:50,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:50,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:51,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:51,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:51,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:52,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:52,002 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:52,002 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:53,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:53,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:53,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:54,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:54,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:54,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:55,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:55,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:55,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:56,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:56,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:56,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:57,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:57,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:57,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:58,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:58,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:58,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:59,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:01:59,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:01:59,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:01,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:01,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:02:01,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:03,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:03,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:02:03,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:04,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:04,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:02:04,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:06,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:06,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:02:06,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:07,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:07,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:02:07,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:08,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:08,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:02:08,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:09,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:09,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:02:09,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:10,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:10,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:02:10,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:11,999 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:12,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 08:02:12,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 08:02:12,002 ERROR [se.dig.sdk.add.ser.OrganizationReindexScheduler] () () (vert.x-eventloop-thread-4) OrganizationReindexScheduler failure: io.vertx.core.impl.NoStackTraceThrowable: Timeout

2023-09-08 08:02:12,002 ERROR [se.dig.sdk.add.ser.AddressReindexScheduler] () () (vert.x-eventloop-thread-4) AddressReindexScheduler failure: java.lang.IllegalStateException: io.vertx.core.impl.NoStackTraceThrowable: Timeout
        at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
        at io.smallrye.mutiny.operators.uni.UniOnFailureTransform$UniOnFailureTransformProcessor.onFailure(UniOnFailureTransform.java:54)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forwardResult(UniCreateFromCompletionStage.java:58)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
        at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
        at io.vertx.core.Future.lambda$toCompletionStage$3(Future.java:538)
        at io.vertx.core.impl.future.FutureImpl$3.onFailure(FutureImpl.java:153)
        at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:75)
        at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:230)
        at io.vertx.core.impl.future.Mapping.onFailure(Mapping.java:45)
        at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:75)
        at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:230)
        at io.vertx.core.impl.future.PromiseImpl.tryFail(PromiseImpl.java:23)
        at io.vertx.core.impl.future.PromiseImpl.onFailure(PromiseImpl.java:54)
        at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:43)
        at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:23)
        at io.vertx.sqlclient.impl.pool.SqlConnectionPool$1PoolRequest.lambda$null$1(SqlConnectionPool.java:224)
        at io.vertx.core.net.impl.pool.SimpleConnectionPool$Cancel.run(SimpleConnectionPool.java:676)
        at io.vertx.core.net.impl.pool.CombinerExecutor.submit(CombinerExecutor.java:56)
        at io.vertx.core.net.impl.pool.SimpleConnectionPool.execute(SimpleConnectionPool.java:245)
        at io.vertx.core.net.impl.pool.SimpleConnectionPool.cancel(SimpleConnectionPool.java:638)
        at io.vertx.sqlclient.impl.pool.SqlConnectionPool$1PoolRequest.lambda$onEnqueue$2(SqlConnectionPool.java:221)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:948)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:919)
        at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
        at io.vertx.core.impl.DuplicatedContext.emit(DuplicatedContext.java:158)
        at io.vertx.core.impl.ContextInternal.emit(ContextInternal.java:194)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.run(VertxImpl.java:937)
        at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
        at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: io.vertx.core.impl.NoStackTraceThrowable: Timeout
```

## Quarkus 3.1.1.Final

```log
2023-09-08 07:42:48,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:48,001 DEBUG [org.hib.res.jdb.int.LogicalConnectionManagedImpl] () () (vert.x-eventloop-thread-4) `hibernate.connection.provider_disables_autocommit` was enabled.  This setting should only be enabled when you are certain that the Connections given to Hibernate by the ConnectionProvider have auto-commit disabled.  Enabling this setting when the Connections do not have auto-commit disabled will lead to Hibernate executing SQL operations outside of any JDBC/SQL transaction.
2023-09-08 07:42:48,002 DEBUG [org.hib.orm.sql.exec] () () (vert.x-eventloop-thread-4) Skipping reading Query result cache data: cache-enabled = true, cache-mode = NORMAL
2023-09-08 07:42:48,002 DEBUG [org.hib.orm.results] () () (vert.x-eventloop-thread-4) Initializer list
2023-09-08 07:42:48,002 DEBUG [org.hib.orm.results] () () (vert.x-eventloop-thread-4)     se.digg.sdk.addressbook.db.entity.OrganizationReindex(a) -> ReactiveEntityResultInitializer(se.digg.sdk.addressbook.db.entity.OrganizationReindex(a))@1604287421 (ReactiveSingleTableEntityPersister(se.digg.sdk.addressbook.db.entity.OrganizationReindex))
2023-09-08 07:42:48,002 DEBUG [org.hib.SQL] () () (vert.x-eventloop-thread-4) select o1_0.resource_id,o1_0.reindex_id from organization_reindex o1_0 offset $1 rows fetch first $2 rows only
2023-09-08 07:42:48,003 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:42:48,003 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:48,005 DEBUG [org.hib.res.jdb.int.LogicalConnectionManagedImpl] () () (vert.x-eventloop-thread-4) Initiating JDBC connection release from afterTransaction
2023-09-08 07:42:48,005 DEBUG [org.hib.res.jdb.int.LogicalConnectionManagedImpl] () () (vert.x-eventloop-thread-4) `hibernate.connection.provider_disables_autocommit` was enabled.  This setting should only be enabled when you are certain that the Connections given to Hibernate by the ConnectionProvider have auto-commit disabled.  Enabling this setting when the Connections do not have auto-commit disabled will lead to Hibernate executing SQL operations outside of any JDBC/SQL transaction.
2023-09-08 07:42:48,006 DEBUG [org.hib.orm.sql.exec] () () (vert.x-eventloop-thread-4) Skipping reading Query result cache data: cache-enabled = true, cache-mode = NORMAL
2023-09-08 07:42:48,006 DEBUG [org.hib.orm.results] () () (vert.x-eventloop-thread-4) Initializer list
2023-09-08 07:42:48,006 DEBUG [org.hib.orm.results] () () (vert.x-eventloop-thread-4)     se.digg.sdk.addressbook.db.entity.AddressReindex(a) -> ReactiveEntityResultInitializer(se.digg.sdk.addressbook.db.entity.AddressReindex(a))@1381611444 (ReactiveSingleTableEntityPersister(se.digg.sdk.addressbook.db.entity.AddressReindex))
2023-09-08 07:42:48,006 DEBUG [org.hib.SQL] () () (vert.x-eventloop-thread-4) select a1_0.resource_id,a1_0.reindex_id from address_reindex a1_0 offset $1 rows fetch first $2 rows only
2023-09-08 07:42:49,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:49,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:42:49,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:50,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:50,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:42:50,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:51,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:51,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:42:51,002 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:53,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:53,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:42:53,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:55,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:55,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:42:55,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:56,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:56,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:42:56,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:57,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:57,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:42:57,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:58,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:58,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:42:58,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:59,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:42:59,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:42:59,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:01,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:01,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:01,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:02,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:02,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:02,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:03,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:03,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:03,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:04,999 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:05,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:05,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:06,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:06,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:06,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:07,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:07,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:07,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:08,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:08,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:08,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:10,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:10,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:10,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:11,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:11,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:11,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:12,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:12,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:12,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:14,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:14,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:14,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:15,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:15,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:15,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:16,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:16,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:16,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:17,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:17,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:17,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:18,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:18,001 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:18,001 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:19,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:19,000 INFO  [se.dig.sdk.add.ser.AddressReindexService] () () (vert.x-eventloop-thread-4) ##### processBatch
2023-09-08 07:43:19,000 DEBUG [org.hib.rea.mut.imp.MutinySessionFactoryImpl] () () (vert.x-eventloop-thread-4) No existing open Mutiny.Session was found in the current Vert.x context: opening a new instance
2023-09-08 07:43:19,003 ERROR [se.dig.sdk.add.ser.OrganizationReindexScheduler] () () (vert.x-eventloop-thread-4) OrganizationReindexScheduler failure: io.vertx.core.impl.NoStackTraceThrowable: Timeout

2023-09-08 07:43:19,003 ERROR [se.dig.sdk.add.ser.AddressReindexScheduler] () () (vert.x-eventloop-thread-4) AddressReindexScheduler failure: java.lang.IllegalStateException: io.vertx.core.impl.NoStackTraceThrowable: Timeout
        at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
        at io.smallrye.mutiny.operators.uni.UniOnFailureTransform$UniOnFailureTransformProcessor.onFailure(UniOnFailureTransform.java:54)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forwardResult(UniCreateFromCompletionStage.java:58)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
        at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
        at io.vertx.core.Future.lambda$toCompletionStage$3(Future.java:386)
        at io.vertx.core.impl.future.FutureImpl$3.onFailure(FutureImpl.java:153)
        at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:75)
        at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:230)
        at io.vertx.core.impl.future.Mapping.onFailure(Mapping.java:45)
        at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:75)
        at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:230)
        at io.vertx.core.impl.future.PromiseImpl.tryFail(PromiseImpl.java:23)
        at io.vertx.core.impl.future.PromiseImpl.onFailure(PromiseImpl.java:54)
        at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:43)
        at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:23)
        at io.vertx.sqlclient.impl.pool.SqlConnectionPool$1PoolRequest.lambda$null$1(SqlConnectionPool.java:224)
        at io.vertx.core.net.impl.pool.SimpleConnectionPool$Cancel.run(SimpleConnectionPool.java:676)
        at io.vertx.core.net.impl.pool.CombinerExecutor.submit(CombinerExecutor.java:56)
        at io.vertx.core.net.impl.pool.SimpleConnectionPool.execute(SimpleConnectionPool.java:245)
        at io.vertx.core.net.impl.pool.SimpleConnectionPool.cancel(SimpleConnectionPool.java:638)
        at io.vertx.sqlclient.impl.pool.SqlConnectionPool$1PoolRequest.lambda$onEnqueue$2(SqlConnectionPool.java:221)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:948)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:919)
        at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
        at io.vertx.core.impl.DuplicatedContext.emit(DuplicatedContext.java:158)
        at io.vertx.core.impl.ContextInternal.emit(ContextInternal.java:194)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.run(VertxImpl.java:937)
        at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
        at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: io.vertx.core.impl.NoStackTraceThrowable: Timeout

2023-09-08 07:43:20,002 ERROR [se.dig.sdk.add.ser.OrganizationReindexScheduler] () () (vert.x-eventloop-thread-4) OrganizationReindexScheduler failure: io.vertx.core.impl.NoStackTraceThrowable: Timeout

2023-09-08 07:43:20,002 ERROR [se.dig.sdk.add.ser.AddressReindexScheduler] () () (vert.x-eventloop-thread-4) AddressReindexScheduler failure: java.lang.IllegalStateException: io.vertx.core.impl.NoStackTraceThrowable: Timeout
        at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
        at io.smallrye.mutiny.operators.uni.UniOnFailureTransform$UniOnFailureTransformProcessor.onFailure(UniOnFailureTransform.java:54)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forwardResult(UniCreateFromCompletionStage.java:58)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
        at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
        at io.vertx.core.Future.lambda$toCompletionStage$3(Future.java:386)
        at io.vertx.core.impl.future.FutureImpl$3.onFailure(FutureImpl.java:153)
        at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:75)
        at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:230)
        at io.vertx.core.impl.future.Mapping.onFailure(Mapping.java:45)
        at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:75)
        at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:230)
        at io.vertx.core.impl.future.PromiseImpl.tryFail(PromiseImpl.java:23)
        at io.vertx.core.impl.future.PromiseImpl.onFailure(PromiseImpl.java:54)
        at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:43)
        at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:23)
        at io.vertx.sqlclient.impl.pool.SqlConnectionPool$1PoolRequest.lambda$null$1(SqlConnectionPool.java:224)
        at io.vertx.core.net.impl.pool.SimpleConnectionPool$Cancel.run(SimpleConnectionPool.java:676)
        at io.vertx.core.net.impl.pool.CombinerExecutor.submit(CombinerExecutor.java:56)
        at io.vertx.core.net.impl.pool.SimpleConnectionPool.execute(SimpleConnectionPool.java:245)
        at io.vertx.core.net.impl.pool.SimpleConnectionPool.cancel(SimpleConnectionPool.java:638)
        at io.vertx.sqlclient.impl.pool.SqlConnectionPool$1PoolRequest.lambda$onEnqueue$2(SqlConnectionPool.java:221)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:948)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.handle(VertxImpl.java:919)
        at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
        at io.vertx.core.impl.DuplicatedContext.emit(DuplicatedContext.java:158)
        at io.vertx.core.impl.ContextInternal.emit(ContextInternal.java:194)
        at io.vertx.core.impl.VertxImpl$InternalTimerHandler.run(VertxImpl.java:937)
        at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
        at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: io.vertx.core.impl.NoStackTraceThrowable: Timeout
```

**Comment by DavideD** [comment]
Why are you using `.subscribe` in the `runTask` method?

Doesn't this work?
```java
@Scheduled(delayed = "10s", every = "{reindex.addresses.schedule.every}")
public Uni<Void> runTask() {
        log.debugv("AddressReindexScheduler start");
        return reindexService
            .processBatch()
            .onItemOrFailure().call(
                item -> log.debugv("AddressReindexScheduler finished: {0}", item.size()),
                failure ->  log.error("AddressReindexScheduler failure", failure)
            )
            .replaceWithVoid();
}
```

**Comment by DavideD** [comment]
> I have also tried to upgrade to quarkus 3.1.1.Final since that contains the 2.0.0.Final release of hibernate reactive but to no avail.

Could you also try with Quarkus 3.3.2 with Hibernate Reactive 2.0.5.Final, please?

**Comment by agreedSkiing** [comment]
> Why are you using `.subscribe` in the `runTask` method?
> 
> Doesn't this work?
> 
> ```java
> @Scheduled(delayed = "10s", every = "{reindex.addresses.schedule.every}")
> public Uni<Void> runTask() {
>         log.debugv("AddressReindexScheduler start");
>         return reindexService
>             .processBatch()
>             .onItemOrFailure().call(
>                 item -> log.debugv("AddressReindexScheduler finished: {0}", item.size()),
>                 failure ->  log.error("AddressReindexScheduler failure", failure)
>             )
>             .replaceWithVoid();
> }
> ```

I have done that change on another branch that hasn't been merged into our main branch yet.

And I will give Quarkus 3.3.2

**Comment by DavideD** [comment]
@dmytro-kostiushko By the way, when using Quarkus, it's better to [create an issue for the Quarkus  repository](https://github.com/quarkusio/quarkus/issues). If you could also provide a project showing the error that we can run, it will also speed up the resolution of the issue.

**Comment by DavideD** [comment]
> .onItemOrFailure().call(
                item -> log.debugv("AddressReindexScheduler finished: {0}", item.size()),
                failure ->  log.error("AddressReindexScheduler failure", failure)
            )

I've just realized I made a minor mistake, it should be `.invoke` (and not `.call`)

**Comment by agreedSkiing** [comment]
So.... after some more testing on 3.3.2 whilst using 5 seconds instead of 1 second between the cronjob triggers and leaving the manuel starts of the cronjob methods as is everything is more stable. It's hasen't solved the issue complelty since I have gotten the timeout errors once or twice but not as frequent as before at least when I was running it manually or not manually but triggering the pipline that runs our test.

I’m also now verifying a change I just pushed which is to assert that our flyway migration is done before each new test that modifies data starts, this is something I previously tried when cronjob triggers being at 1 second but at that point the tests failed with the timeout exceptions again :(. 

Now so 6 out of 30 runs in (thanks god that I did a curl bash script to trigger the pipeline more autmaticly) it seems to be very stable.

**Comment by agreedSkiing** [comment]
Okay, that took some time and in total I had 5 out 30 runs fail because of the timemout issue.

**Comment by agreedSkiing** [comment]
I will see what if I can create a reproducer project, but can't promise anything

**Comment by DavideD** [comment]
Actually, I think I know what's going on.

Could you try the following code for getBatch?:

```java
    public Uni<List<AddressReindex>> getBatch(int max) {
        return sf.openSession()
				.chain( session -> session
						.createQuery(  "FROM AddressReindex", AddressReindex.class )
                                                .setMaxResults(max)
                                                .getResultList()
						.eventually( session::close )
				)
            .onFailure()
            .transform(IllegalStateException::new);
    }
```

**Comment by DavideD** [comment]
Actually, something like this is safer:
```java
    public Uni<List<AddressReindex>> getBatch(int max) {
        return sf.openSession()
				.chain( session -> Uni.createFrom().item( () -> session
						.createQuery(  "FROM AddressReindex", AddressReindex.class )
                                                .setMaxResults(max)
                                                .getResultList() )
						.eventually( session::close )
				)
            .onFailure()
            .transform(IllegalStateException::new);
    }
```

I think what's going on is that if the jobs are scheduled too close to each other the session gets reused.
That's because `withSession` doesn't create a new session each time. `openSession` does.

**Comment by agreedSkiing** [comment]
But that code would create a Uni<Uni<List<AddressReindex>> since it uses `Uni.createFrom()`

**Comment by DavideD** [comment]
Ah, right. I think using `Uni.createFrom().deferred` should work.

**Comment by agreedSkiing** [comment]
Yes that worked better, and I have been thinking the same thig that 
> I think what's going on is that if the jobs are scheduled too close to each other the session gets reused.

**Comment by DavideD** [comment]
If these changes make it work, I'm going to close this issue.

We are aware of these type of issues and how hard is to spot them unless one is familiar with the use case (like in this case). We have some ideas for a solution, but we are still in the brainstorming/prototyping phase

**Comment by agreedSkiing** [comment]
Went around in the code and changed all the `sf.withSession()` to `sf.openSession()` and I'm still getting the timeout error, but there are also some `sf.withTransaction()` that I don't know how to convert... so I instead wrote a test that just stressed the `getBatch` operation with 1 nanosecond delay between each call thanks to Awaitility and when using `sf.openSession()` it works fine, but when I use the old code I'm instead getting another error (see below) and I don't think that error is relevant here. Anyway, that ´getBatch´ won't be called with that short of an intervall and I think my problem or my team’s problem is that our tests are doing somethings, like calling flyway to recreate the database, that won't be happening in a production enviroment and that me and my teammates have not been able to recreate the timeout exception with our lightwieght curl stress testing, when running quarkus in production mode in our kubernetes enviroment.
<details>
	<summary>New and probably irrelevant error</summary>

```log
2023-09-22 08:06:35,467 ERROR [se.dig.sdk.add.ser.AddressReindexScheduler] () () (vert.x-eventloop-thread-4) AddressReindexScheduler failure: io.smallrye.mutiny.CompositeException: Multiple exceptions caught:
        [Exception 0] java.lang.IllegalStateException: The current operation requires a safe (isolated) Vert.x sub-context, but the current context hasn't been flagged as such. You can still use Hibernate Reactive, you just need to avoid using the methods which implicitly require accessing the stateful context, such as MutinySessionFactory#withTransaction and #withSession.
        [Exception 1] java.lang.IllegalStateException: The current operation requires a safe (isolated) Vert.x sub-context, but the current context hasn't been flagged as such. You can still use Hibernate Reactive, you just need to avoid using the methods which implicitly require accessing the stateful context, such as MutinySessionFactory#withTransaction and #withSession.
        at io.smallrye.mutiny.operators.uni.UniOnTermination$UniOnTerminationProcessor.onFailure(UniOnTermination.java:49)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
        at io.smallrye.mutiny.operators.uni.UniOnItemConsume$UniOnItemComsumeProcessor.invokeEventHandler(UniOnItemConsume.java:82)
        at io.smallrye.mutiny.operators.uni.UniOnItemConsume$UniOnItemComsumeProcessor.onItem(UniOnItemConsume.java:42)
        at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem$KnownItemSubscription.forward(UniCreateFromKnownItem.java:38)
        at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem.subscribe(UniCreateFromKnownItem.java:23)
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
        at io.smallrye.mutiny.operators.uni.UniOnItemConsume.subscribe(UniOnItemConsume.java:30)
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
        at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
        at io.smallrye.mutiny.operators.uni.UniOnTermination.subscribe(UniOnTermination.java:21)
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
        at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureFlatMap.subscribe(UniOnItemOrFailureFlatMap.java:27)
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
        at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:81)
        at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
        at io.smallrye.mutiny.operators.uni.UniOnItemTransform$UniOnItemTransformProcessor.onItem(UniOnItemTransform.java:43)
        at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onItem(UniOperatorProcessor.java:47)
        at io.smallrye.mutiny.operators.uni.builders.UniCreateFromItemSupplier.subscribe(UniCreateFromItemSupplier.java:29)
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
        at io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap.subscribe(UniOnFailureFlatMap.java:31)
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
        at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:81)
        at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
        at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onItem(UniOperatorProcessor.java:47)
        at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forwardResult(UniCreateFromCompletionStage.java:63)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
        at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
        at io.vertx.core.Future.lambda$toCompletionStage$3(Future.java:536)
        at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
        at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
        at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
        at io.vertx.core.impl.future.Mapping.onSuccess(Mapping.java:40)
        at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
        Suppressed: java.lang.IllegalStateException: The current operation requires a safe (isolated) Vert.x sub-context, but the current context hasn't been flagged as such. You can still use Hibernate Reactive, you just need to avoid using the methods which implicitly require accessing the stateful context, such as MutinySessionFactory#withTransaction and #withSession.
                at io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.checkIsSafe(VertxContextSafetyToggle.java:89)
                at io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.validateContextIfExists(VertxContextSafetyToggle.java:72)
                at io.quarkus.hibernate.reactive.runtime.customized.CheckingVertxContext.remove(CheckingVertxContext.java:52)
                at org.hibernate.reactive.mutiny.impl.MutinySessionFactoryImpl.lambda$withSession$19(MutinySessionFactoryImpl.java:212)
                at io.smallrye.context.impl.wrappers.SlowContextualRunnable.run(SlowContextualRunnable.java:19)
                at io.smallrye.mutiny.groups.UniOnTerminate.lambda$invoke$0(UniOnTerminate.java:51)
                at io.smallrye.mutiny.operators.uni.UniOnTermination$UniOnTerminationProcessor.onFailure(UniOnTermination.java:47)
                ... 45 more
Caused by: java.lang.IllegalStateException: The current operation requires a safe (isolated) Vert.x sub-context, but the current context hasn't been flagged as such. You can still use Hibernate Reactive, you just need to avoid using the methods which implicitly require accessing the stateful context, such as MutinySessionFactory#withTransaction and #withSession.
        at io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.checkIsSafe(VertxContextSafetyToggle.java:89)
        at io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.validateContextIfExists(VertxContextSafetyToggle.java:72)
        at io.quarkus.hibernate.reactive.runtime.customized.CheckingVertxContext.put(CheckingVertxContext.java:40)
        at org.hibernate.reactive.mutiny.impl.MutinySessionFactoryImpl.lambda$withSession$17(MutinySessionFactoryImpl.java:210)
        at io.smallrye.mutiny.groups.UniOnItem.lambda$invoke$0(UniOnItem.java:58)
        at io.smallrye.context.impl.wrappers.SlowContextualConsumer.accept(SlowContextualConsumer.java:21)
        at io.smallrye.mutiny.operators.uni.UniOnItemConsume$UniOnItemComsumeProcessor.invokeEventHandler(UniOnItemConsume.java:77)
        ... 43 more
```
</details>

**Comment by DavideD** [comment]
You don't need to change all the `withSession` to `openSession`. It was just in this particular case, because you are using a scheduler and I don't think you have an alternative if you risk to have operation in parallel.

In general, you should use `.withSession` and `.withTransaction`. These methods do more than just creating the session and make it safer to use Hibernate Reactive with Quarkus.

But all these are Quarkus usability issues, I will close this one and I would suggest to open a new issue on the [Quarkus issue tracker](https://github.com/quarkusio/quarkus/issues), now that you have a better idea about what's going on.
